src/components/about-us/section-03/index.js

```diff
@@ -298,24 +297,66 @@ export default class Section3 extends PureComponent {
     })
   }
 
-  /**
-   *  Given data list and return the max and min year of current data
-   *  @param {Array} list
-   *  @returns {Array} [maxYear, minYear]
-   */
+  _getConfig = () => {
+    return axios.get(configUrl)
+      .then(res => {
+        this.setState({ config: _.get(res, 'data.rows') })
+      })
+      .catch((err) => {
+        console.error(err) // eslint-disable-line no-console
+        logger.errorReport({
+          report: err,
+          message: 'Something went wrong during getting configs for about-us page section3'
+        })
+      })
+  }
+
   _getYearRange = (list) => {
-    return list.map(item => Number(item.date.split('/')[0])).reduce((accumulator, currentValue) => {
-      return [
-        (!accumulator[0]) ? currentValue : Math.min(currentValue, accumulator[0]),
-        (!accumulator[1]) ? currentValue : Math.max(currentValue, accumulator[1])
-      ]
-    },[])
+    if (list) {
+      const allYears = _.keys(
+        _.groupBy(list, award => {
+          if (award.date) {
+            return award.date.split('/')[0]
+          }
+        })
+      )
+      return (
+        <YearRange>
+          {Math.min(...allYears)}-{Math.max(...allYears)}
+        </YearRange>
+      ) 
+    }
   }
 
   render() {
-    const { activeYearIndex, activeAwardIndex, activeAwardId } = this.state
-    const currentYear = awardYearList[activeAwardIndex][activeYearIndex]
-    const selectedDataList = awardGroupByNameAndYear[activeAwardIndex][currentYear]
+    const { config, activeYearIndex, activeAward } = this.state
+    const groupedByAward = _.groupBy(config, award => award['award.zh-tw'])
+
+    // The `groupedByAwardAndYer` object will be like:
+    // {
+    //   "award1": {
+    //     "2017": [ {...} ]
+    //   },
+    // }
+    let groupedByAwardAndYear = {}
+
+    // The `awardYearInOrder` object will be like:
+    // {
+    //   "award1": [ "2019", "2018" ]  // list years in descending order
+    // }
+    let awardYears = {}
+
+    _.keys(groupedByAward).map((key) => {
+      groupedByAwardAndYear[key] = _.groupBy(
+        groupedByAward[key], 
+        record => record.date.split('/')[0]
+      )
+      awardYears[key] = _.orderBy(_.keys(groupedByAwardAndYear[key])).reverse()
+    })
+    
+    const currentYear = _.get(awardYears, `${activeAward}.${activeYearIndex}`)
+    const selectedRecords = _.get(groupedByAwardAndYear, `${activeAward}.${currentYear}`, [])
+
```

Comment:
Two minor suggestions.

1. Could `awardYears` and `groupedByAwardAndYear`  be extracted out of render function? Therefore, we don't need to build them in every rendering.

2. How about providing `awardYears` and `groupedByAwardAndYear` directly in the config ? 
We have a program to transfer spreadsheet data into current data structure specified in `section3.master.json`, how about adding a further step to transfer them into `awardYears` and `groupedByAwardAndYear` data structure?
That would be much clearer for understanding the data structure for me.

Reply:
The program will build `awardYears` and `groupedByAwardAndYear` in the existing [config](https://about-us.twreporter.org/section3.master.json), like suggestion 2 said.

Please see commit [1166ba2](https://github.com/twreporter/twreporter-react/pull/1652/commits/1166ba289b5d03ba03e9c3b3803939c67e8dd4f8), thanks.